src/lib/parser.rs

```diff
@@ -109,10 +109,9 @@ impl<'a, TokId: Clone + Copy + Debug + PartialEq + TryFrom<usize> + TryInto<usiz
         let mut tstack: Vec<Node<TokId>> = Vec::new();
         let mut errors: Vec<ParseError<TokId>> = Vec::new();
         let accpt = psr.lr(0, &mut pstack, &mut tstack, &mut errors);
-        let t = tstack.drain(..).nth(0).unwrap();
```

Comment:
The commit message seems to suggest this changes some behaviour but the code just inlines a variable and removes the variable declaration. What exactly changes here?

Reply:
If no repairs are found, `tstack` is empty, so taking the 0th element causes a panic (empty list). The killer line in the commit is line 114:

```rust
            (true, false)  => Err((Some(tstack.drain(..).nth(0).unwrap()), errors)),
```

which means "if we found repairs that allowed us to fully parse the file, still return a parse tree". Line 116 (in the diff) says "if we didn't find a successful parse *and* we didn't find repairs, don't return a parse tree".